packages/flutter/lib/src/widgets/focus_scope.dart

```diff
@@ -1043,3 +1044,35 @@ class ExcludeFocus extends StatelessWidget {
     );
   }
 }
+
+/// A focus area is used to expand a focus node as part of the render object layer
+/// outside of its descendant render objects.
+///
+/// This is used by the [TextField] widgets to associate the focus node with the input
+/// decorations.
+class FocusArea extends SingleChildRenderObjectWidget {
```

Comment:
This widget can really only work with the FocusTrap, It feel weird that one is private and another is public. I feel we should either make both public or both private. Curious what others think? @gspencergoog